agent/libs/platform.lua

```diff
@@ -431,6 +431,73 @@ if ffi.os ~= "Windows" then
 
 end
 
+
+local function attachReader(stream, onData, onError)
+  local function onEvent(err, data)
+    return coroutine.wrap(function()
+      if err then
+        return onError(err)
+      end
+      stream:read_stop()
+      onData(data)
+      stream:read_start(onEvent)
+    end)()
+  end
+  stream:read_start(onEvent)
+end
+
+local function makeWriter(stream, onError)
+  return function (chunk)
+    local err
+    if chunk then
+      err = async(stream.write, stream, chunk)
+    else
+      err = async(stream.shutdown, stream)
+      if not stream:is_closing() then
+        stream:close()
+      end
+    end
+    if err then
+      onError(err)
+    end
+  end
+end
+
+local function makeKiller(child, handles)
+  return function (signal)
+    child:kill(signal)
+    if not child:is_closing() then
+      child:close()
+    end
+    for i = 1, #handles do
+      if not handles[i]:is_closing() then
+        handles[i]:close()
+      end
+    end
+  end
+end
+
+function platform.spawn(command, options, onStdout, onStderr, onError, onExit)
+
+  local stdin = uv.new_pipe(false)
+	local stdout = uv.new_pipe(false)
+	local stderr = uv.new_pipe(false)
```

Comment:
check the indentation here too.  You can probably just replace tabs with two spaces in the entire file.